File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/NavigationViewModel.java
package com.dhu.dhu_timetable.ui.navitem;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.dhu.dhu_timetable.model.LoginModel;
import com.dhu.dhu_timetable.repo.UsersRepo;

public class NavigationViewModel extends ViewModel {
    MutableLiveData<LoginModel> test;
    UsersRepo usersRepo;
    String TAG = "janghee";

    public NavigationViewModel() {
        if (test != null) {
            return;
        }
        usersRepo = UsersRepo.getInstance();
    }

    public void init(String email) {
        test = usersRepo.getUser(email);
        Log.d(TAG, "init: ");
    }

    public MutableLiveData<LoginModel> getTest() {
        return test;
    }
}

File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/NavigationViewModel.kt
package com.dhu.dhu_timetable.ui.navitem

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.dhu.dhu_timetable.model.LoginModel
import com.dhu.dhu_timetable.repo.UsersRepo
import com.dhu.dhu_timetable.repo.UsersRepo.Companion.getInstance
import java.lang.Exception

class NavigationViewModel : ViewModel() {
    var loginModel: MutableLiveData<LoginModel>? = null
    private lateinit var usersRepo: UsersRepo
    private var TAG = "janghee"

    fun initLoginData(email: String?) {
        loginModel = usersRepo.getUser(email!!)
        Log.d(TAG, "init: ")
    }

    init {
        try {
            if (loginModel == null) {
                usersRepo = getInstance()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}